python/torch_mlir/csrc/base_lazy_backend/dynamic_ir.cpp

```diff
@@ -0,0 +1,74 @@
+//===- dynamic_ir.cpp -----------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+// This file is adapted from pytorch/pytorch
+// https://github.com/pytorch/pytorch/blob/master/torch/csrc/lazy/ts_backend/dynamic_ir.cpp
+//===----------------------------------------------------------------------===//
+
+#include <torch/csrc/lazy/ts_backend/dynamic_ir.h>
+
+namespace torch {
+namespace lazy {
+
+DimensionNode::DimensionNode(OpKind op, OpList operands, hash_t hash_seed)
```

Comment:
Is there any doc on the strategy here for dynamic shapes?

Reply:
There is, but its very much a work in progress I believe. This class is just a placeholder introduced for now. We'll need to make further changes to this once they finalize the design and implemention in PyTorch master.